os_stub/cryptlib_openssl/pk/x509.c

```diff
@@ -2151,3 +2161,336 @@ bool libspdm_x509_get_cert_from_cert_chain(const uint8_t *cert_chain,
+    case LIBSPDM_CRYPTO_NID_RSASSA4096:
+    case LIBSPDM_CRYPTO_NID_RSAPSS4096:
+        ret = EVP_PKEY_set1_RSA(private_key, (RSA *)context);
+        if (ret != 1) {
+            goto free_all;
+        }
+        break;
+    case LIBSPDM_CRYPTO_NID_ECDSA_NIST_P256:
+    case LIBSPDM_CRYPTO_NID_ECDSA_NIST_P384:
+    case LIBSPDM_CRYPTO_NID_ECDSA_NIST_P521:
+        ret = EVP_PKEY_set1_EC_KEY(private_key, (EC_KEY *)context);
+        if (ret != 1) {
+            goto free_all;
+        }
+        break;
+    default:
+        goto free_all;
+    }
+
+    /* requester info parse*/
+    if (requester_info_length != 0)
+    {
+        result1 = libspdm_get_obj_by_oid_from_req_info(requester_info, requester_info_length,
+                                                       challenge_password_buffer,
+                                                       &challenge_password_size,
+                                                       m_libspdm_oid_challenge_password,
+                                                       sizeof(m_libspdm_oid_challenge_password));
+        if (result1 == 1) {
+            ret = X509_REQ_add1_attr_by_NID(x509_req, NID_pkcs9_challengePassword,
+                                            V_ASN1_UTF8STRING, challenge_password_buffer,
+                                            challenge_password_size);
+            if (ret == 0) {
+                LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO,"set challenge Password error\n"));
+                return false;
+            }
+        } else if (result1 == -1) {
+            return false;
+        }
+
+        result2 = libspdm_get_obj_by_oid_from_req_info(requester_info, requester_info_length,
+                                                       extension_request_buffer,
+                                                       &extension_request_size,
+                                                       m_libspdm_oid_extension_request,
+                                                       sizeof(m_libspdm_oid_extension_request));
+        if (result2 == 1) {
+            ret = X509_REQ_add1_attr_by_NID(x509_req, NID_ext_req,
+                                            V_ASN1_UTF8STRING, extension_request_buffer,
+                                            extension_request_size);
+            if (ret == 0) {
+                LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO,"set extension request error\n"));
+                return false;
+            }
+        } else if (result2 == -1) {
+            return false;
+        }
+
+        /*the req info is invalid*/
+        if ((result1 == 0) && (result2 == 0)) {
+            return false;
+        }
+    }
+
+    /*set version of x509 req*/
+    ret = X509_REQ_set_version(x509_req, version);
+    if (ret != 1) {
+        LIBSPDM_DEBUG((LIBSPDM_DEBUG_INFO,"set version error\n"));
+        goto free_all;
+    }
+
+    /*set subject of x509 req*/
+    x509_name = X509_REQ_get_subject_name(x509_req);
+
+    if (orgnization != NULL) {
```

Comment:
I dont understand how does this work?

orgnization is assigned to NULL in the beginning. And never assigned to a real value.